### PR TITLE
fix(weave): Make traces work again and still remove logging

### PR DESCRIPTION
**Pull request by jwlee64**
main change is to patch_all except logging
and than patch logging if allowed

I also changes the os.getenv to compare against true, vs just checking for existence, it made testing and turning the flag off easier

### CODE REVIEW

File: weave/engine_trace.py [hunk 1]
```diff
@@ -155,23 +155,23 @@ def attributes(self):
     def set_tag(self, key, unredacted_val, redacted_val=None):
         if "tags" not in self.attributes:
             self.attributes["tags"] = {}
-        if not os.getenv("DISABLE_WEAVE_PII"):
+        if not os.getenv("DISABLE_WEAVE_PII") == "true":
```
Comment (nfoucha) [inline weave/engine_trace.py:158]: Nit: Maybe a `lower()` and a `strip()` call on the getenv would be a good idea for input safety.

Comment (jamie-rasmussen) [reply weave/engine_trace.py:158]: I think creating a helper method may be the way to go.

Comment (jwlee64) [reply weave/engine_trace.py:158]: yup making a helper in environment.py

File: weave/weave_server.py [hunk 2]
```diff
@@ -71,8 +71,13 @@ def execute(self, document, *args, **kwargs):
     # crashes
     import ddtrace
 
-    if not os.environ.get("DISABLE_WEAVE_PII"):
-        ddtrace.patch_all(logging=True)
+    # patch all libraries to datadog except logging
```
Comment (jwlee64) [inline weave/weave_server.py:74]: the main fix is here

**Review by tssweeney — APPROVED** [review]

**Pull request merged**